tools/build_chain.sh

```diff
@@ -1391,13 +1526,19 @@ download_bin()
     chmod a+x ${bin_path}
 }
 
+function version_gt() { test "$(echo "$@" | tr " " "\n" | sort -V | head -n 1)" != "$1"; }
+
 check_bin()
 {
     echo "Checking fisco-bcos binary..."
-    bin_version=$(${bin_path} -v)
-    if [ -z "$(echo ${bin_version} | grep 'FISCO-BCOS')" ];then
+    bin_version=$()
+    if ! ${bin_path} -v | grep -q 'FISCO-BCOS';then
         exit_with_clean "${bin_path} is wrong. Please correct it and try again."
     fi
+    bin_version=$(${bin_path} -v | grep -oe "[2-9]*\.[0-9]*\.[0-9]*" )
+    if version_gt ${compatibility_version} ${bin_version};then
```

Comment:
Codacy found an issue: [Double quote to prevent globbing and word splitting.](https://app.codacy.com/gh/FISCO-BCOS/FISCO-BCOS/pullRequest?prid=5859193)